planner/core/exhaust_physical_plans.go

```diff
@@ -588,7 +588,19 @@ func (p *LogicalJoin) constructInnerIndexScan(ds *DataSource, idx *model.IndexIn
 
 	is.initSchema(ds.id, idx, cop.tablePlan != nil)
 	indexConds, tblConds := splitIndexFilterConditions(filterConds, idx.Columns, ds.tableInfo)
-	path := &accessPath{indexFilters: indexConds, tableFilters: tblConds, countAfterIndex: math.MaxFloat64}
+	path := &accessPath{
+		indexFilters:     indexConds,
+		tableFilters:     tblConds,
+		countAfterAccess: rowCount,
+	}
+	if len(indexConds) > 0 {
+		selectivity, _, err := ds.tableStats.HistColl.Selectivity(ds.ctx, indexConds)
+		if err != nil {
+			logutil.Logger(context.Background()).Warn("calculate selectivity faild, use selection factor", zap.Error(err))
+			selectivity = selectionFactor
+		}
+		path.countAfterIndex = rowCount * selectivity
```

Comment:
How about make a new method on `path` for line `596~603` and `exhaust_physical_plans.go:506~513`.

Reply:
Hmm, these lines are simple enough and are unnecessary to be extracted?